src/Kerbero.Infrastructure/NukiActions/Repositories/NukiSmartLockExternalRepository.cs

```diff
@@ -12,21 +13,21 @@ namespace Kerbero.Infrastructure.NukiActions.Repositories;
 
 public class NukiSmartLockExternalRepository: INukiSmartLockExternalRepository
 {
+	private readonly NukiExternalRepositoryHelper _repositoryHelper;
 	private readonly NukiExternalOptions _options;
-	private readonly ILogger<NukiSmartLockExternalRepository> _logger;
 
-	public NukiSmartLockExternalRepository(IOptions<NukiExternalOptions> options, 
-		ILogger<NukiSmartLockExternalRepository> logger)
+	public NukiSmartLockExternalRepository(IOptions<NukiExternalOptions> options, NukiExternalRepositoryHelper repositoryHelper)
 	{
+		_repositoryHelper = repositoryHelper;
 		_options = options.Value;
-		_logger = logger;
 	}
 	
 	public async Task<Result<List<NukiSmartLockExternalResponse>>> GetNukiSmartLocks(string accessToken)
 	{
-		var apiResponse = await $"{_options.BaseUrl}"
+		var apiResponse = await _repositoryHelper.SafeHttpCall<List<NukiSmartLockExternalResponse>>( async () => await $"{_options.BaseUrl}"
```

Comment:
```suggestion
		var apiResponse = await _repositoryHelper.SafeHttpCall<List<NukiSmartLockExternalResponse>>( () => $"{_options.BaseUrl}"
```
cosi dovrebbe andare lo stesso =)